studio/src/aem/content-navigation.js

```diff
@@ -64,11 +72,63 @@ class ContentNavigation extends LitElement {
         this.unregisterFromSource();
     }
 
+    toggleFilterPanel() {
+        this.showFilterPanel = !this.showFilterPanel;
+    }
+
     registerToSource() {
         this.source = document.getElementById(this.getAttribute('source'));
         if (!this.source) return;
         this.source.addEventListener(EVENT_LOAD, this.forceUpdate);
         this.source.addEventListener(EVENT_CHANGE, this.forceUpdate);
+        this.source.getTopFolders().then((folders) => {
+            this.topFolders = folders;
+        });
+    }
+
+    selectConsumer(consumer) {
+        if (!consumer) return;
+        this.source.path = consumer;
+        this.source.openFolder(consumer);
+        this.source.searchFragments();
+    }
+
+    handleConsumerChange(event) {
+        this.selectConsumer(event.target.value);
+    }
+
+    get consumers() {
```

Comment:
name suggestions: `get roots`

Reply:
I found consumer too opinionated, because, in my idea, everything under aem folder should remain generic and reusable.

Reply:
i can keep top folders

Reply:
but i guess i like the binding of top folders to consumers/surfaces: i agree that nala/docs/drafts surfaces are strange, but we could still consider them this way.
Essentially they are top folders with limited write access, and that is how we designed our consumer/surface storage